configure.ac

```diff
@@ -1,18 +1,46 @@
 
 AC_PREREQ([2.69])
-AC_INIT([ledmon], [0.96])
+AC_INIT([ledmon], [0.96.0])
 AC_CONFIG_MACRO_DIR([m4])
 AC_SUBST([PACKAGE_DATE], "May 2022")
 AM_INIT_AUTOMAKE([-Wall -Werror foreign])
 
+LT_PREREQ([2.4.6])
+AM_PROG_AR
+LT_INIT
+
 # Checks for programs.
 
 AX_CHECK_PROG([gcc])
 AX_CHECK_PROG([make])
 
-AM_PROG_CC_C_O
-AC_PROG_CC_C99
+
+# Using the version number we will construct the appropriate libtool version.
+# Following semantic versioning (X.Y.Z) we will bump X when we have an incompatible
+# library version, bump Y when we add a new public function to the library and, bump
+# Z when we fix a bug/improve the library internals. We can only add to library, we
+# can never remove for backwards compatibility
+# eg.
+# start 0.96.0
+# Add a new library function led_foo -> 0.97.0
+# Fix a bug in library -> 0.97.1
+# Fix another bug -> 0.97.2
+# Add another new library function -> 0.98.0
+
+LIBLED_MAJOR_VERSION=`echo $VERSION | awk -F. '{print $1}'`
```

Comment:
IMHO we should have 1 source for the version, so that they cannot mistakenly differ.  We could keep major/minor/micro separate, but then we would need to reconstruct the version string.  Would that be more acceptable to you?

Reply:
> IMHO we should have 1 source for the version, so that they cannot mistakenly differ

Good point but having the variables should be also safe enough.
I checked following libs:
- https://github.com/libvirt/libvirt/blob/master/meson.build#L161 - string is built but it uses meson,
- https://github.com/GNOME/libxml2/blob/master/configure.ac - automake, it uses variables,
- https://github.com/intel/libxcam/blob/master/configure.ac - automake, it uses variables,
- https://github.com/besser82/libxcrypt/blob/develop/configure.ac - automake, version variable set in AC_INIT

I cannot determine which approach is better but I would prefer to have variable for each version part.

> We could keep major/minor/micro separate, but then we would need to reconstruct the version string

Could you please visualize your idea. Do you plan to do something like here?
```
LIBXML_MAJOR_VERSION=MAJOR_VERSION
LIBXML_MINOR_VERSION=MINOR_VERSION
LIBXML_MICRO_VERSION=MICRO_VERSION
LIBXML_MICRO_VERSION_SUFFIX=
LIBXML_VERSION=$LIBXML_MAJOR_VERSION.$LIBXML_MINOR_VERSION.$LIBXML_MICRO_VERSION$LIBXML_MICRO_VERSION_SUFFIX
LIBXML_VERSION_INFO=`expr $LIBXML_MAJOR_VERSION + $LIBXML_MINOR_VERSION`:$LIBXML_MICRO_VERSION:$LIBXML_MINOR_VERSION
```